### PR TITLE
Uninstalling if cost of repair is larger than cost of content

### PR DESCRIPTION
**Pull request by Dzejkop**

### CODE REVIEW

File: Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs [hunk 1]
```diff
@@ -100,11 +100,15 @@ AppContentSummary latestVersionContentSummary
             if (repairCost < contentSize)
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs]: repairCost seems to be a number of missing and broken files (x2). Is contentSize the content size in bytes?
This does not seem right.

Comment (Dzejkop) [reply Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs]: Below that there is:
```csharp
if (isNewVersionAvailable)
{
    repairCost *= latestVersionContentSummary.Chunks.Size;
}
else
{
    repairCost *= installedVersionContentSummary.Chunks.Size;
}
```

repairCost is estimated to be equal to the sum of missing and broken files, multiplied by two and then multiplied by chunk size in the respective version.

Comment (genail) [reply Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs]: Sorry, I didn't notice that.
I can't imagine it though. For instance if there's a large file broken, would the cost be the same as for small file?

Comment (Dzejkop) [reply Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs]: I changed it now. The cost calculation will now use actual file sizes.

File: Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs [hunk 2]
```diff
@@ -112,6 +105,13 @@ AppContentSummary latestVersionContentSummary
             }
         }
 
+        private long CalculateRepairCost(AppContentSummary contentSummary, IEnumerable<FileIntegrity> filesToRepair)
+        {
+            return filesToRepair
+                .Select(f => contentSummary.Files.FirstOrDefault(e => e.Path == f.FileName))
+                .Sum(f => f.Size);
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs]: Will it work? Select() transforms enumerable to a new form (https://msdn.microsoft.com/en-us/library/bb548891(v=vs.110).aspx). When a file is not found, a null is returned. Sum() is dereferencing the value so it looks like NRE for me.

Comment (witcher112) [reply Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs]: When a file is not found, default value of struct `AppContentSummaryFile` is returned. There won't be any NRE here.

**Pull request merged**